widgets/var_widgets.py

```diff
@@ -272,17 +272,17 @@ def insert(self, *a, **kw):
         to_track = []
 
         try:
-            values = kw["values"]
+            old_values = kw.pop("values")
         except KeyError:
             pass
         else:
-            if not isinstance(values, (tuple, list)):
-                values = (values,)
+            kw["values"] = values = []
 
-            for col, v in enumerate(values):
+            for col, v in enumerate(old_values):
                 if isinstance(v, variables):
                     to_track.append((col, v))
-                    values[col] = v.get()
+                    v = v.get()
+                values.insert(col, v)
```

Comment:
Maybe:
```
        try:
	    kw["values"] = values = list(kw.pop("values"))
        except KeyError:
            pass
        else:
            for col, v in enumerate(values):
                if isinstance(v, variables):
                    to_track.append((col, v))
                    values[col] = v.get()
```

Reply:
Changing list during iteration may raise an exception. Did you tried this snippet?

Reply:
I ran the GUI and received no exceptions.